lib/ClangImporter/ImportName.cpp

```diff
@@ -999,7 +999,11 @@ bool NameImporter::hasNamingConflict(const clang::NamedDecl *decl,
   lookupResult.setAllowHidden(true);
   lookupResult.suppressDiagnostics();
 
-  if (clangSema.LookupName(lookupResult, /*scope=*/clangSema.TUScope)) {
+  // Only force the Objective-C codepath in LookupName if clangSema.TUScope is
+  // nullptr
+  if (clangSema.LookupName(lookupResult, /*scope=*/clangSema.TUScope,
+                           /*AllowBuiltinCreation=*/false,
+                           /*ForceNoCPlusPlus=*/!clangSema.TUScope)) {
```

Comment:
Thinking about this further, instead of setting this to `!clangSema.TUScope` that `hasNamingConflict` could have a bool flag to forward here. That way we could control which calls to `hasNamingConflict` would use C/ObjC LookupName codepath. What do y'all think?